core/citrus-api/src/test/java/org/citrusframework/context/TestContextUnitTest.java

```diff
@@ -0,0 +1,87 @@
+package org.citrusframework.context;
+
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.report.TestListener;
+import org.citrusframework.report.TestListeners;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+public class TestContextUnitTest {
+
+    private TestContext fixture;
+
+    @BeforeTest
+    void beforeTestClass() {
+        fixture = new TestContext();
+    }
+
+    @Test
+    public void handleErrorGracefullyHandlesErrorInTestStartListener() {
+        var testListenerMock = fixtureWithTestListeners();
+
+        var cause = new CitrusRuntimeException("thrown with a purpose!");
+        doThrow(cause).when(testListenerMock).onTestStart(any(TestContext.EmptyTestCase.class));
+
+        invokeHandleErrorOnFixture(cause);
+
+        verify(testListenerMock).onTestStart(any(TestContext.EmptyTestCase.class));
+        verifyNoMoreInteractions(testListenerMock);
+    }
+
+    @Test
+    public void handleErrorGracefullyHandlesErrorInTestFailureListener() {
+        var testListenerMock = fixtureWithTestListeners();
+
+        var cause = new CitrusRuntimeException("thrown with a purpose!");
+        doThrow(cause).when(testListenerMock).onTestFailure(any(TestContext.EmptyTestCase.class), any(CitrusRuntimeException.class));
+
+        invokeHandleErrorOnFixture(cause);
+
+        verify(testListenerMock).onTestStart(any(TestContext.EmptyTestCase.class));
+        verify(testListenerMock).onTestFailure(any(TestContext.EmptyTestCase.class), any(CitrusRuntimeException.class));
+        verifyNoMoreInteractions(testListenerMock);
+    }
+
+    @Test
+    public void handleErrorGracefullyHandlesErrorInTestFinishListener() {
+        var testListenerMock = fixtureWithTestListeners();
+
+        var cause = new CitrusRuntimeException("thrown with a purpose!");
+        doThrow(cause).when(testListenerMock).onTestFinish(any(TestContext.EmptyTestCase.class));
+
+        invokeHandleErrorOnFixture(cause);
+
+        verify(testListenerMock).onTestStart(any(TestContext.EmptyTestCase.class));
+        verify(testListenerMock).onTestFailure(any(TestContext.EmptyTestCase.class), any(CitrusRuntimeException.class));
+        verify(testListenerMock).onTestFinish(any(TestContext.EmptyTestCase.class));
+    }
+
+    private TestListener fixtureWithTestListeners() {
```

Comment:
Just to add something to this review:
the name of this method implies, that it returns the fixture wired with listeners. Instead it does exactly this, but returns the listeners.

So maybe use: attachTestListenerMockToFixture